File: Iniciante/1046 - Tempo de Jogo/TypeScript/1046.ts
import * as fs from "fs";

const TOTAL_HOURS_IN_A_DAY: number = 24;

function main(): void {
  const input: string = fs.readFileSync("/dev/stdin", "utf8");
  const [startTime, finalTime] = input.trim().split(/\s+/).map(Number);
  const gameDuration: number = calculateGameDuration(startTime, finalTime);
  console.log(`O JOGO DUROU ${gameDuration} HORA(S)`);
}

function calculateGameDuration(startTime: number, finalTime: number): number {
  if (startTime > finalTime) {
    return finalTime + TOTAL_HOURS_IN_A_DAY - startTime;
  } else if (startTime === finalTime) {
    return TOTAL_HOURS_IN_A_DAY;
  } else {
    return finalTime - startTime;
  }
}

main();

File: Iniciante/1046 - Tempo de Jogo/Python/1046.py
TOTAL_HOURS_IN_A_DAY = 24


def main():
    start_time, final_time = map(int, input().split())
    game_duration = calculate_game_duration(start_time, final_time)
    print(f"O JOGO DUROU {game_duration} HORA(S)")


def calculate_game_duration(start_time, final_time):
    if start_time > final_time:
        return (final_time + TOTAL_HOURS_IN_A_DAY) - start_time
    elif (start_time == final_time):
        return TOTAL_HOURS_IN_A_DAY
    else:
        return final_time - start_time


if __name__ == "__main__":
    main()